go/vt/workflow/long_polling.go

```diff
@@ -157,9 +158,48 @@ func getID(url, base string) (int, error) {
 	return i, nil
 }
 
+const (
+	LogLevelDumpAllHTTPHeaders glog.Level = 50
```

Comment:
Ohh, using the loglevel to determine behavior is very elegant! 

Another way to do this that you may have already considered is to add a flag. Recently, I came across https://github.com/vitessio/vitess/pull/9550 by @mattlord, which is similar in spirit in that it redacts sensitive information from the logs (bind variables, in this particular case). That PR introduces the [`sanitize_log_messages` flag](https://github.com/vitessio/vitess/pull/9550/files#diff-ee5ffc675b719e69de7147068ab4ecf03f6d5db91a000d212a30f8ce61544691R167) on the `vttablet` for this purpose. 

I'd bias towards using a flag, since it's explicitly opt-in, and I can see an (admittedly imaginary) scenario where one would want debug logging on but to still redact the headers. However! I acknowledge there are definitely trade-offs in adding a flag. (vtctlds already have a billion flags, for example.) 

I don't feel very strongly about this, but I'm curious what you think. :) 

(@mattlord maybe you have an opinion too, since you've been thinking about this for https://github.com/vitessio/vitess/issues/9469 🙇 )

Reply:
Yeah, I think that adding a flag with the same name and general behavior makes sense for `vtctld` as well (I think we'll eventually add this to `vtgate` too). Thank you for working on this! ❤️

Reply:
sounds good, it's good to be consistent with the rest of the project in how these toggles are implemented.  Let me see how to wire that up with a flag 🚧

Reply:
any objections on making this new flag ( `dump_all_http_headers` ) default's value = `false`?

Reply:
most likely it'll end-up being `workflow_manager_sanitize_http_headers` with the default value = `true`

how does that sound?

Reply:
... with the goal to make default behavior safe out of the box

Reply:
Hmmm I do like making things safer by default, but because this is arguably a breaking change, it might be easier to start with a default of `workflow_manager_sanitize_http_headers=false`. That way the behaviour is entirely opt-in. This is what https://github.com/vitessio/vitess/pull/9550 does.

Either way, I would also suggest adding this to the release notes, which will be for Vitess 14.0, unless you feel strongly about backporting this to the current Vitess 13.0 release candidate. (FWIW it'll make no difference to Slack as we'll have to cherry pick this change to our fork anyhow, given that we're not yet ready for 13.0.) I don't think the release notes for 14.0 are being drafted yet, but I'll tag this PR with "release notes" so we don't forget! :)

Reply:
Also, with respect to the flag name of `workflow_manager_sanitize_http_headers`, it might be fine to make this flag a bit more general, which is again what https://github.com/vitessio/vitess/pull/9550 does with its `sanitize_log_messages` flag. We could even use the same name, as @mattlord suggests.

Reply:
ok this is done